### PR TITLE
Show latest config in /v1/agent/self

### PR DESCRIPTION
**Pull request by nvanthao**
This is a manual backport of: https://github.com/hashicorp/consul/pull/18681

### CODE REVIEW

File: agent/config/deep-copy.sh [hunk 1]
```diff
@@ -0,0 +1,13 @@
+#!/usr/bin/env bash
+# Copyright (c) HashiCorp, Inc.
+# SPDX-License-Identifier: BUSL-1.1
```
Comment (dhiaayachi) [inline agent/config/deep-copy.sh:3]: This need to be removed for backports

**Review by dhiaayachi — CHANGES_REQUESTED** [review]
Thank you @nvanthao 
I left a comment about stripping the license.

File: agent/config/deep-copy.sh (hunk 1 above)
Comment (nvanthao) [reply agent/config/deep-copy.sh]: Apologies @dhiaayachi I missed this. This has been done 🙇

**Review by dhiaayachi — APPROVED** [review]
LGMT! 🚀

**Pull request merged**